lib/lrmd/lrmd_client.c

```diff
@@ -197,14 +197,8 @@ lrmd_new_event(const char *rsc_id, const char *task, guint interval_ms)
     lrmd_event_data_t *event = calloc(1, sizeof(lrmd_event_data_t));
 
     CRM_ASSERT(event != NULL);
-    if (rsc_id != NULL) {
-        event->rsc_id = strdup(rsc_id);
-        CRM_ASSERT(event->rsc_id != NULL);
-    }
-    if (task != NULL) {
-        event->op_type = strdup(task);
-        CRM_ASSERT(event->op_type != NULL);
-    }
+    pcmk__copy_or_null((char **) &event->rsc_id, rsc_id);
```

Comment:
pcmk__copy_or_null doesn't write around in strings. so the prototype could be a pointer to const char *.

Reply:
lrmd\_event\_data\_t wrongly declares its dynamically allocated strings as const char \* which means they can't be freed without casting. So no matter how we declare the new function, that struct will need casting.